packages/cli/src/deploy/ensureWorldFactory.ts

```diff
@@ -1,111 +1,34 @@
-  const initModuleBytecode = encodeDeployData({
-    bytecode: initModuleBuild.bytecode.object as Hex,
-    abi: initModuleAbi,
-    args: [accessManagementSystem, balanceTransferSystem, batchCallSystem, registration],
-  });
-
-  const initModule = getCreate2Address({ from: deployerAddress, bytecode: initModuleBytecode, salt });
-
-  const build = withWorldProxy ? worldProxyFactoryBuild : worldFactoryBuild;
-  const abi = withWorldProxy ? worldProxyFactoryAbi : worldFactoryAbi;
-
-  const worldFactoryDeployedBytecodeSize = size(build.deployedBytecode.object as Hex);
-  const worldFactoryBytecode = encodeDeployData({
-    bytecode: build.bytecode.object as Hex,
-    abi: abi,
-    args: [initModule],
-  });
-
-  const worldFactory = getCreate2Address({ from: deployerAddress, bytecode: worldFactoryBytecode, salt });
-
-  const worldFactoryContracts: readonly Contract[] = [
-    {
-      bytecode: accessManagementSystemBytecode,
-      deployedBytecodeSize: accessManagementSystemDeployedBytecodeSize,
-      label: "access management system",
-    },
-    {
-      bytecode: balanceTransferSystemBytecode,
-      deployedBytecodeSize: balanceTransferSystemDeployedBytecodeSize,
-      label: "balance transfer system",
-    },
-    {
-      bytecode: batchCallSystemBytecode,
-      deployedBytecodeSize: batchCallSystemDeployedBytecodeSize,
-      label: "batch call system",
-    },
-    {
-      bytecode: registrationBytecode,
-      deployedBytecodeSize: registrationDeployedBytecodeSize,
-      label: "core registration system",
-    },
-    {
-      bytecode: initModuleBytecode,
-      deployedBytecodeSize: initModuleDeployedBytecodeSize,
-      label: "core module",
-    },
-    {
-      bytecode: worldFactoryBytecode,
-      deployedBytecodeSize: worldFactoryDeployedBytecodeSize,
-      label: "world factory",
-    },
-  ];
-
-  // WorldFactory constructor doesn't call InitModule, only sets its address, so we can do these in parallel since the address is deterministic
-  await ensureContractsDeployed({
-    client,
-    deployerAddress,
-    contracts: worldFactoryContracts,
-  });
-
-  return worldFactory;
+  if (!withWorldProxy) {
+    const contracts = getWorldFactoryContracts(deployerAddress);
+
+    // WorldFactory constructor doesn't call InitModule, only sets its address, so we can do these in parallel since the address is deterministic
+    await ensureContractsDeployed({
+      client,
+      deployerAddress,
+      contracts: Object.values(contracts),
+    });
+
+    return contracts.WorldFactory.address;
+  } else {
+    const contracts = getWorldProxyFactoryContracts(deployerAddress);
+
+    // WorldFactory constructor doesn't call InitModule, only sets its address, so we can do these in parallel since the address is deterministic
+    await ensureContractsDeployed({
+      client,
+      deployerAddress,
+      contracts: Object.values(contracts),
+    });
+
+    return contracts.WorldProxyFactory.address;
+  }
```

Comment:
I like to return early in these cases

```suggestion
  if (withWorldProxy) {
    const contracts = getWorldProxyFactoryContracts(deployerAddress);
    // WorldFactory constructor doesn't call InitModule, only sets its address, so we can do these in parallel since the address is deterministic
    await ensureContractsDeployed({
      client,
      deployerAddress,
      contracts: Object.values(contracts),
    });
    return contracts.WorldProxyFactory.address;
  }
  
  const contracts = getWorldFactoryContracts(deployerAddress);
  // WorldFactory constructor doesn't call InitModule, only sets its address, so we can do these in parallel since the address is deterministic
  await ensureContractsDeployed({
    client,
    deployerAddress,
    contracts: Object.values(contracts),
  });
  return contracts.WorldFactory.address;
```